### ISSUE TITLE
Pyscripter Will Show Python As a Registered Device but will not connect to it

### ISSUE THREAD

**Issue by Meg720** [pyscripter/pyscripter#957]
PyScripter 3.6 x86 will not connect to Python 3.7 (32 bit). It will show Python 3.7 (32 bit) as a registered device, but says "Python Not Available." I have read through a lot of chats and haven't found the answer. Can anyone help?

**Comment by pyscripter** [comment]
Which distribution of python are you using?  Have a look at #938 as well.

**Comment by Meg720** [comment]
Python 3.7.3 Release Date March 25, 2019 "Windows x86 Executable Installer."  From this website: https://www.python.org/downloads/release/python-373/

**Comment by Meg720** [comment]
Also, I checked out #938 and #881 I am not using Anaconda. I am using python.org to download. I added Python to the Unregistered section even though it was showing as a Registered product as well and it did not work for me.

**Comment by Meg720** [comment]
Also, I uninstalled the 32bit of both products and downloaded the 64bit versions. I was unable to get to "Hello World" with the 64bit. When I uninstalled the 64bit versions, I and reinstalled the 32bit versions, "Hello World" was not working either. I assumed I had messed with the PATH variable too much and perhaps there were some left over parts installed from Python that were messing up the connection. So, I did a factory reset on the computer I am using and then tried reinstalling Python and PyScripter at 32bit. Instead of manually messing with the PATH variable, I just selected the "Add to PATH" option on the install, and everything works fine with Python (version above) connecting to the Command Prompt again after the factory reset, but still cannot connect to PyScripter despite showing as a registered device. Here are some images in case they are helpful (let me know if anything needs clarification): 
![Capture1](https://user-images.githubusercontent.com/51974025/59776740-4eae6800-9281-11e9-9289-3ae8ecc8b239.PNG)
![Capture](https://user-images.githubusercontent.com/51974025/59776756-579f3980-9281-11e9-9422-ad51c9dc0873.PNG)
![Capture2](https://user-images.githubusercontent.com/51974025/59776920-a1881f80-9281-11e9-8d8a-59114f5716df.PNG)

**Comment by omatsei** [comment]
I'm having the same problem. Fresh install of Windows 10, everything 64-bit. I have both Python 2.7.16 and Python 3.7.3 installed, but PyScripter refuses to use 3.7.3. It lists it as a Registered Version too, same as the pic above.

I also tried making a batch file and specifying the PYTHONHOME, and using every combination of "--python37" and "--pythondllpath" I could think of (independently, both at the same time, and pointing to both the direct DLL file itself and the folder it's contained in). Nothing works so far.

**Comment by omatsei** [comment]
I just uninstalled everything Python related, and re-installed using the 32-bit versions. 2.7.16 is under C:\Python27, and 3.7.3 is under C:\Python37. PyScripter is under C:\Program Files (x86).

Everything was downloaded from python.org, nothing from Anaconda or Miniconda. Both versions were installed for all users. Nothing added to any environment variable or path, but 3.7.3 might register itself when it's installed, I don't know.

I have 2 batch files for each version, both using the variation of "PYTHONHOME=C:\PythonXX" depending on their version. The 2.7 version works perfectly. The 3.7 version does not work at all. 

PyScripter sees both versions as Unregistered, but only 3.7 as registered. When I select 3.7 and try to activate it, it says "PyScripter could not load a Python engine".

**Comment by omatsei** [comment]
For troubleshooting purposes, I removed 3.7.3 and installed 3.7.2, and it WORKS. 

So yeah, there's definitely a bug with PyScripter not detecting 3.7.3.

**Comment by pyscripter** [comment]
Tried python 3.7.3 64 bits with PyScripter 3.6 64 bits and  it works.
![Capture](https://user-images.githubusercontent.com/1311616/59887262-5843e700-93cb-11e9-8598-fe07e01f149e.JPG)
Will try 32 bits next.

**Comment by omatsei** [comment]
I’m running as a regular user, not a user with admin rights, if that helps.

**Comment by pyscripter** [comment]
32 bits works fine as well:
![Capture](https://user-images.githubusercontent.com/1311616/59888474-9d1e4c80-93d0-11e9-91dc-8a974b893976.JPG)

I see your point about admin rights but I cannot see why 3.7.2 works fine and 3.7.3 doesn't.  PyScripter does not normally need admin rights.   Have you used the same installation options with both versions?

**Comment by omatsei** [comment]
I tried it on a brand new system, running Windows 10 LTSC. The exact commands I used to install everything were:

`msiexec /i python-2.7.16.msi InstallAllUsers=1 TARGETDIR="C:\Python27" /qn /norestart`
`python-3.7.3.exe /quiet InstallAllUsers=1 TargetDir="c:\Python37"`
(or)
`python-3.7.2.exe /quiet InstallAllUsers=1 TargetDir="c:\Python37"`
`PyScripter-3.6.0-x86-Setup.exe /SILENT /VERYSILENT`

I'm also creating some batch files to start PyScripter with both versions of Python, and putting them in the same directory as PyScripter. Those batch files are:

pyscripter-python27.bat:
`SET PYTHONHOME="C:\Python27"`
`"C:\Program Files (x86)\PyScripter\PyScripter" --PYTHON27 --PYTHONDLLPATH "C:\Python27" %1 %2 %3 %4 %5`

pyscripter-python37.bat:
`SET PYTHONHOME="C:\Python37"`
`"C:\Program Files (x86)\PyScripter\PyScripter" --PYTHON37 --PYTHONDLLPATH "C:\Python37" %1 %2 %3 %4 %5`

I'm also putting shortcuts in the Start Menu for each of those batch files, but I doubt anything in those shortcuts would have anything to do with it (it doesn't work when I launch the batch files from a command prompt anyway).

If you can see anything I'm doing differently, or incorrectly, please let me know. I have to replicate this setup on 250 computers this summer, so I want it to work as well as possible.

**Comment by pyscripter** [comment]
So 3.7.2 works and 3.7.3 doesn't!!  Most wierd.

I cannot see anything wrong.  However if you do not have admin rights it may not be a good idea to use InstallAllUsers.   Or else if you run the installer, there is an option to precompile the standard libraries (there may be a command line flag for that).  Give it a try.

**Comment by pyscripter** [comment]
250 computers! By the way, what is the organisation you are working for?

**Comment by omatsei** [comment]
I'm installing everything with admin rights, but running it as a standard user.

I work for the College of IST at Penn State University. We're in the process of rebuilding all of our computer labs this summer, in preparation for classes in the fall.

**Comment by omatsei** [comment]
After a little more testing, I discovered that it works after installing Anaconda and rebooting. Anaconda does add a few directories to the PATH... could that be it, even though Anaconda has 3.7.3 and PyScripter is using 3.7.2?

**Comment by pyscripter** [comment]
Did you register Anaconda?
What is the registry value of 
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\PythonComputer\HKEY_CURRENT_USER\Software\Python\PythonCore\3.7-32\InstallPath  

Are you using the bat file with the PYTHONDLLPATH?

**Comment by omatsei** [comment]
There's no HKEY_LOCAL_MACHINE\SOFTWARE\PythonComputer or HKEY_CURRENT_USER\Software\Python

The Python that comes with Anaconda is not registered. 

Yes, the bat file has both PYTHONHOME set to either C:\Python27 or C:\Python37, and PYTHONDLLPATH set to each.

**Comment by omatsei** [comment]
I just tried it with 3.7.3 and it works... I don't know what was going on yesterday with it. Sorry for the wild goose chase.

**Comment by Meg720** [comment]
OP here, after reading @omatsei  success with 3.7.2, I downloaded it just to see if it would work on my computer. When I reopened PyScripter, it had connected to the previously installed 3.7.3. So, I uninstalled 3.7.2, and 3.7.3 is still connected to PyScripter. After a basic print test, everything seems to be working fine! In hindsight, in all the troubleshooting, I am not sure if I ever just closed out of PyScripter and reopened it, but wanted to share my experience for anyone else having similar issues. Thank you both for your help.
